File: frontend/lib/models/episode.dart
class Episode {
  final int id;
  final int storyId;
  final String title;
  final String audioUrl;
  final int? duration;
  final int episodeNumber;

  Episode({
    required this.id,
    required this.storyId,
    required this.title,
    required this.audioUrl,
    this.duration,
    required this.episodeNumber,
  });

  factory Episode.fromJson(Map<String, dynamic> json) {
    return Episode(
      id: json['id'],
      storyId: json['storyId'],
      title: json['title'],
      audioUrl: json['audioUrl'],
      duration: json['duration'],
      episodeNumber: json['episodeNumber'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'storyId': storyId,
      'title': title,
      'audioUrl': audioUrl,
      'duration': duration,
      'episodeNumber': episodeNumber,
    };
  }

  String get formattedDuration {
    if (duration == null) return '--:--';
    final minutes = duration! ~/ 60;
    final seconds = duration! % 60;
    return '${minutes.toString().padLeft(2, '0')}:${seconds.toString().padLeft(2, '0')}';
  }
}

File: backend/src/main/java/com/audiostory/model/Episode.java
package com.audiostory.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "episodes")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Episode {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "story_id", nullable = false)
    private Story story;
    
    @Column(nullable = false)
    private String title;
    
    @Column(name = "audio_url", nullable = false)
    private String audioUrl;
    
    private Integer duration; // in seconds
    
    @Column(name = "episode_number", nullable = false)
    private Integer episodeNumber;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
